apis/v1alpha1/gatewayclass_types.go

```diff
@@ -168,6 +168,12 @@ type GatewayClassStatus struct {
 	// +optional
 	// +kubebuilder:default={{type: "InvalidParameters", status: "Unknown", message: "Waiting for controller", reason: "Waiting", lastTransitionTime: "1970-01-01T00:00:00Z"}}
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+	// ProvisionedGateways is a list of Gateways that have been provisioned
+	// using this class. Implementations must add any Gateways of this class to
+	// this list once they have been provisioned and remove Gateways as soon as
+	// they are deleted or deprovisioned.
+	ProvisionedGateways []GatewayObjectReference `json:"provisionedGateways"`
```

Comment:
Questions: Is is possible for different controllers to race here?

Reply:
I don't think so. Each GatewayClass should only be managed by a single controller.